core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -278,9 +279,18 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
           // we only call wakeup() once to close the consumer,
           // so if we catch it in commit we can safely retry
           // and re-throw to break the loop
-          consumerWrapper.commit()
+          CoreUtils.swallow(consumerWrapper.commit(), this)
           throw e
 
+        case _: TimeoutException =>
+          Try(consumerWrapper.consumer.listTopics) match {
+            case Success(visibleTopics) =>
+              consumerWrapper.offsets.retain((tp, _) => visibleTopics.containsKey(tp.topic))
+            case _ =>
+          }
+          warn("Failed to commit offsets because the offset commit request processing can not be completed in time. " +
```

Comment:
Related to the concern of data loss, would it be useful to also keep track and log the time of last successful commit in the warning message, so that SRE can gauge how much time worth of data has been duplicated?